spec/javascript/packages/document-capture/hooks/use-log-camera-info-spec.tsx

```diff
@@ -0,0 +1,115 @@
+import sinon from 'sinon';
+import { render } from '@testing-library/react';
+import { useLogCameraInfo } from '@18f/identity-document-capture/hooks/use-log-camera-info';
+import { AnalyticsContextProvider } from '@18f/identity-document-capture/context/analytics';
+import userEvent from '@testing-library/user-event';
+
+interface MockComponentProps {
+  isBackOfId: boolean;
+  hasStartedCropping: boolean;
+}
+
+function MockComponent({ isBackOfId, hasStartedCropping }: MockComponentProps) {
+  useLogCameraInfo({ isBackOfId, hasStartedCropping });
+  return <div>mockcomponent</div>;
+}
+
+const mockTrack = {
+  label: 'mockTrackLabel',
+  getSettings: () => ({
+    frameRate: 60,
+    height: 3000,
+    width: 3000,
+  }),
+};
+
+const mockDeviceId = 'mockDeviceId';
+const mockGetVideoTracks = () => [mockTrack];
+const mockDevice = {
+  kind: 'videoinput',
+  deviceId: mockDeviceId,
+};
+
+const mockGetUserMediaThrowsError = () => {
+  throw new Error('camera logging failed');
+};
+const mockGetUserMedia = () =>
+  new Promise((resolve) => resolve({ getVideoTracks: mockGetVideoTracks }));
+const mockEnumerateDevices = () => new Promise((resolve) => resolve([mockDevice]));
+
+// Give the global navigator object a mock camera with various functions so that the logging actually happens
+const addMockCameraToNavigator = () => {
+  Object.defineProperty(global.navigator, 'mediaDevices', {
+    value: {
+      getUserMedia: mockGetUserMedia,
+      enumerateDevices: mockEnumerateDevices,
+    },
+    configurable: true,
+    writable: true,
+  });
+};
+
+describe('document-capture/hooks/use-log-camera-info', () => {
+  addMockCameraToNavigator();
```

Comment:
All tests run in the same global scope, so we'll want to tear down anything we set up in the global namespace to avoid leaking into other tests.

For this, I think the `useDefineProperty` helper in the `test-helpers` package might work for what you're doing.

Reply:
Wow, that's [so much better](https://github.com/18F/identity-idp/pull/10227/commits/de0c5a5e1f727f88c5e5a8bb68ab8367718cf076). Thanks!